app/controllers/host_aggregate_controller.rb

```diff
@@ -27,10 +27,11 @@ def new
     assert_privileges("host_aggregate_new")
     @host_aggregate = HostAggregate.new
     @in_a_form = true
-    @ems_choices = {}
-    Rbac::Filterer.filtered(ManageIQ::Providers::CloudManager).select { |ems| ems.supports?(:create_host_aggregate) }.each do |ems|
-      @ems_choices[ems.name] = ems.id
-    end
+
+    supported_types = HostAggregate.descendants.select { |klass| klass.supports?(:create) }.map(&:module_parent).map(&:name)
```

Comment:
As we start going more in this direction I think a helper method like `subclass_supports?(:create)` would help here.
Asking "what types of X support Y and what are their equivalent EMS names" seems like it'd be a pretty common thing to do /cc @kbrock

Reply:
I think @kbrock had a similar proposal.

Reply:
Okay I don't want to overload this PR too much, I'll open a follow-up on core to add something like this

Reply:
All that non leaf and openstack changes. Not sure if we can just do `descendants` here.

At first, I thought this would be very easy to extract `subclasses_supports?(feature)` and move from `ExtManagementSystem` to `SupportsFeatureMixin` but `concrete_subclasses` (specifically the references to `permitted?` and `leaf_subclasses`) makes it a little more tricky.

Maybe the concept of `supported_descendants` can be put in there and that is overridden by ems/others that need special logic.

Aside: `subclasses_supports?(feature)` should probably not have a question mark on it. (it is not used in many places so that would not be so bad to change before it becomes too rampant)

Reply:
So I figured `descendants` would be okay because the `supports :create` only will be true on the "concrete_subclasses" equivalent (aka `Openstack::CloudManager::HostAggregate` defines `supports :create` and `PowerVC::CloudManager::HostAggregate` inherits that)

Reply:
Example:
```
>> HostAggregate.descendants.select { |klass| klass.supports?(:create) }.map(&:name)
=> ["ManageIQ::Providers::Openstack::CloudManager::HostAggregate", "ManageIQ::Providers::IbmPowerVc::CloudManager::HostAggregate"]
```

Reply:
https://github.com/ManageIQ/manageiq/pull/21553

```
HostAggregate.types_supporting(:create)
```

Hmm... Not working for me right now. guessing `supports :create` is not defined over there in my repo

Reply:
Ohhh I like `types_sypporting`...good name.